content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -22,21 +22,131 @@ external components communicate with one another.
+
+[The Discovery API](#discovery-api) provides information about the Kubernetes APIs: API names,
+resources, versions, and supported operations. This is a Kubernetes specific term as it is a separate API from the Kubernetes OpenAPI.
+It is intended to be a small document providing a summary of available resources
+and does not detail specific schema for the resources. For reference around schemas
+for resources, please refer to the OpenAPI document.
+
+The [Kubernetes OpenAPI Document]{#openapi-specification} provides (full) [OpenAPI v2.0 and 3.0 schemas](https://www.openapis.org/) for all Kubernetes API
+endpoints. OpenAPI v3 is the preferred method for accessing OpenAPI as it
+provides a more comprehensive and accurate view of the API. It includes all
+available API paths, as well as all resources consumed and produced for every
+operation on every endpoint. This also includes any extensibility components
+that a cluster supports. This document a complete document and is significantly larger than Discovery. The Discovery API is a
+lightweight alternative that provides a subset of information in the
+Kubernetes OpenAPI document.
+
+## Discovery API
+
+Kubernetes publishes a list of all group versions and resources supported via
+the Discovery API. This includes the following for each resource:
+
+- Name
+- Cluster or Namespace scope
+- Endpoint URL and supported verbs
+- Alternative names
+- Group, version, kind
+
+The API is available both aggregated and unaggregated form. The aggregated
+discovery serves two endpoints while the unaggregated discovery serves a
+separate endpoint for each group version.
 
-Consider using one of the [client libraries](/docs/reference/using-api/client-libraries/)
-if you are writing an application using the Kubernetes API.
+### Aggregated Discovery
+
+{{< feature-state state="beta"  for_k8s_version="v1.27" >}}
+
+Kubernetes offers beta support for aggregated discovery, publishing
+all resources supported by a cluster through two endpoints (`/api` and
+`/apis`). Requesting this
+endpoint drastically reduces the number of requests sent to fetch the
+discovery for the average Kubernetes cluster. This may be accessed by
+requesting the respective endpoints with an Accept header indicating
+the aggregated discovery resource:
+`Accept: application/json;v=v2beta1;g=apidiscovery.k8s.io;as=APIGroupDiscoveryList`.
+
+Without the Accept header resource type explicitly indicated, the default
+response for the `/api` and `/apis` endpoint is an unaggregated discovery
+document.
+
+The [discovery document](https://github.com/kubernetes/kubernetes/blob/release-v{{< skew currentVersion >}}/api/discovery/aggregated_v2beta1.json) serving built-in resources is found in the Kubernetes Github repository. This Github document can be used as a reference of the base set of available resources if a Kubernetes cluster is not available to query.
+
+The endpoint also supports ETag and protobuf encoding.
+
+### Unaggregated discovery
```

Comment:
nit: Instead of "Unaggregated" how about "Navigable discovery" or something?

Reply:
I slightly prefer aggregated because throughout this entire doc (and codebase) we've been referring to "aggregated discovery" and "unaggregated discovery". Navigable discovery feels like it could be a new orthogonal feature and I'd prefer to avoid that ambiguity.